lib/linearizer.js

```diff
@@ -137,11 +139,20 @@ module.exports = class Linearizer {
 
     // get the indexed nodes
     const indexed = []
-    while (true) {
-      const nodes = this.consensus.shift()
-      if (!nodes.length) break
 
-      this._yield(nodes, indexed)
+    // apply consensus heads jit to avoid batch sizes affecting it
+    for (let i = 0; i < this._pendingConsensusHeads.length; i++) {
```

Comment:
I fail to see a flow where `this._pendingConsensusHeads` does not have length 0 or 1, so not sure why there's a for loop.

`this._pendingConsensusHeads` is only modified in this method, which is fully sync.

Before the for loop, a node can get added to `_pendingConsensusHeads`

Then there's the for loop, which is all sync, and then `this._pendingConsensusHeads = []` is set to the empty array again.

So next time this method runs, `this._pendingConsensusHeads` will just be the empty array at the start.

Reply:
Oops, I had missed the indent of the surrounding for loop, then it makes sense indeed